watchface

```diff
@@ -0,0 +1,406 @@
+-q or --qemu    communicates with QEMU emulated watch (same as -r localhost -p 2222 )
+-w or --wall WP sets the wallpaper for deploy or test to the named file
+
+Available commands:
+version         displays the version of this program and exits
+deploy WF       pushes the named watchface to the watch and activates it
+clone WF NEWWF  clones the named watchface WF to new watchface NEWWF
+test WF         sets the active watchface to the named watchface
+
+EOF
+}
+
+function doWatchCommand {
+    local user="$1"
+    local cmd=$2
+    case ${user} in
+        root)
+            if [ "$ADB" = "true" ] ; then
+                adb shell "${cmd}"
+            else
+                ssh -p "${WATCHPORT}" -t root@"${WATCHADDR}" ${cmd}
+            fi
+            ;;
+        ceres)
+            if [ "$ADB" = "true" ] ; then
+                printf -v cmd %q "${cmd}"
+                adb shell "su -l -c ${cmd} ceres"
+            else
+                ssh -p "${WATCHPORT}" -t ceres@"${WATCHADDR}" ${cmd}
+            fi
+            ;;
+        *)
+            echo "Error: unknown watch user ${user}"
+            ;;
+    esac
+}
+
+function setDconf {
+    local dconfsetting="$1"
+    local filename="$2"
+    doWatchCommand "ceres" "dconf write ${dconfsetting} '\"file://${filename}\"'"
+}
+
+function pushFiles {
+    local user="$1"
+    local sourcedir="$2"
+    local destdir="$3"
+    if [ "$ADB" = "true" ] ; then
+        adb push ${sourcedir} "${destdir}"
+    else
+        scp -P"${WATCHPORT}" -r ${sourcedir} "${user}@${WATCHADDR}:${destdir}"
+    fi
+}
+
+function pullFiles {
+    local user="$1"
+    local sourcedir="$2"
+    local destdir="$3"
+    if [ "$ADB" = "true" ] ; then
+        adb pull ${sourcedir} "${destdir}"
+    else
+        scp -P"${WATCHPORT}" -r "${user}@${WATCHADDR}:${sourcedir}" "${destdir}"
+    fi
+}
+
+function pushWatchface {
+    pushFiles "root" "${1}"'/usr/share/*' "/usr/share/"
+}
+
+function pushWallpaper {
+    local source="$1"
+    local wallpaper
+    wallpaper="$(basename "$1")"
+    local destination="/usr/share/asteroid-launcher/wallpapers/full/${wallpaper}"
+    pushFiles "root" "${source}" "${destination}"
+    setDconf "/desktop/asteroid/background-filename" "${destination}"
+}
+
+function restartCeres {
+    doWatchCommand "root" "systemctl restart user@1000"
+}
+
+function rebootWatch {
+    doWatchCommand "ceres" "reboot"
+}
+
+function activateWatchface {
+    setDconf "/desktop/asteroid/watchface" "/usr/share/asteroid-launcher/watchfaces/${1}.qml"
+}
+
+function setTimeZone {
+    doWatchCommand "root" "timedatectl set-timezone $1"
+}
+
+function forgetKeys {
+    ssh-keygen -R "${WATCHADDR}"
+    ssh-keygen -R watch
+}
+
+function watchfaceClone {
+    local files
+    mapfile -t files <<<"$(find . -path './'"$1"'/*' -type f | awk '{ dst=$0 ; gsub(/'"$1"'/,"mmmm",dst);  print $0 " " dst }')"
+    for myfile in "${files[@]}" ; do
+        IFS=" " read -r -a onefile <<< "${myfile}"
```

Comment:
Here too the "one" prefix doesn't seem to mean much.

Since it's an array containing the source and destination paths, maybe srcDstPaths would be clearer ?

Reply:
Yes, fixed.